scripts/un/sdg/util.py

```diff
@@ -0,0 +1,233 @@
+    'FOOT_NOTE', 'REPORTING_TYPE', 'OBSERVATION_STATUS', 'RELEASE_STATUS',
+    'RELEASE_NAME'
+}
+
+# Create map of M49 -> ISO-alpha3 for countries.
+with open(os.path.join(module_dir_, 'm49.csv')) as f:
+    PLACES = {}
+    reader = csv.DictReader(f, delimiter='\t')
+    for row in reader:
+        if not row['ISO-alpha3 code']:  # Only countries for now.
+            continue
+        PLACES[int(row['M49 code'])] = row['ISO-alpha3 code']
+
+# Create map of name -> dcid for supported cities.
+with open(os.path.join(module_dir_, 'cities.csv')) as f:
+    reader = csv.DictReader(f)
+    CITIES = {row['name']: row['dcid'] for row in reader}
+
+
+def format_description(s):
+    '''Formats input with curated style.
+
+    Args:
+        s: Input string.
+
+    Returns:
+        Curated string.
+    '''
+    # Remove <=2 levels of ().
+    formatted = re.sub('\((?:[^)(]|\([^)(]*\))*\)', '', s)
+    # Remove <=2 levels of [].
+    formatted = re.sub('\[(?:[^)(]|\[[^)(]*\])*\]', '', formatted)
+    # Remove attributes indicated with 'by'.
+    formatted = formatted.split(', by')[0]
+    # Remove references indicated by 'million USD'.
+    formatted = formatted.split(', million USD')[0]
+    # Remove extra spaces
+    formatted = formatted.replace(' , ', ', ').replace('  ', ' ').strip()
+    # Remove trailing commas
+    if formatted[-1] == ',':
+        formatted = formatted[:-1]
+    # Replace 100,000 with 100K
+    formatted = formatted.replace('100,000', '100K')
+    # Make ascii
+    return formatted.replace('Â',
+                             '').replace('’', '\'').replace('₂', '2').replace(
+                                 '\xa0', ' ').replace('−', '-')
+
+
+def is_float(element):
+    '''Checks if value can be interpreted as float.
+
+    Args:
+      element: Input.
+
+    Returns:
+      Whether the value can be cast as a float.
+    '''
+    if element is None:
+        return False
+    try:
+        float(element)
+        return True
+    except ValueError:
+        return False
+
+
+def is_valid(v):
+    '''Checks if value should be included in csv.
+
+    Args:
+      v: Input.
+
+    Returns:
+      Whether the value can be used.
+    '''
+    try:
+        return not math.isnan(float(v))
+    except ValueError:
+        return v and not v == 'nan'
+
+
+def format_variable_description(variable, series):
+    '''Curates variable descriptions.
+
+    Args:
+      variable: Full variable description.
+      series: Series portion of description.
+
+    Returns:
+      Formatted description.
+    '''
+    parts = variable.split(series)
+    return format_description(series) + parts[1] if len(
+        parts) > 1 else format_description(series)
+
+
+def format_variable_code(code):
+    '''Curates variable codes.
+
+    Args:
+      code: Input code.
+
+    Returns:
+      Formatted code.
+    '''
+    parts = code.split('?')
+    return parts[0] + ':' + parts[1].replace('=', '--').replace(
```

Comment:
To confirm, are we allowing `:` in var?

Reply:
I just picked these since the linter was okay with them, but it's quick to change to a different format if preferred!

Reply:
[from offline discussion] the new format will be `sdg/<CODE>~<PROPERTY1>-<VALUE1>.<PROPERTY2>-<VALUE2>`
thanks!!